### PR TITLE
model-app

### PR DESCRIPTION
**Pull request by ThangPhanCong**

### CODE REVIEW

File: app/models/relationship.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
```
Comment (thinhnv58) [inline app/models/relationship.rb]: "User" -> :User

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_many :microposts
+  has_one :profile
+  has_many :likes
+  has_many :comments
+
+  has_many :active_relationships, class_name: "Relationship", foreign_key: "follower_id", dependent: :destroy
```
Comment (thinhnv58) [inline app/models/user.rb]: "Relationship" -> :Relationship

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  has_many :passive_relationships, class_name: "Relationship", foreign_key: "followed_id", dependent: :destroy
```
Comment (thinhnv58) [inline app/models/user.rb]: "Relationship" -> :Relationship

File: app/models/relationship.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+  belongs_to :followed, class_name: "User"
```
Comment (thinhnv58) [inline app/models/relationship.rb]: "User" -> :User

File: app/models/user.rb (hunk 2 above)
Comment (thinhnv58) [inline app/models/user.rb]: foreign_key: :follower_id

File: db/migrate/20170612032128_create_relationships.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateRelationships < ActiveRecord::Migration[5.0]
+  def change
+    create_table :relationships do |t|
+      t.integer :follower_id
+      t.integer :followed_id
+
+
```
Comment (thinhnv58) [inline db/migrate/20170612032128_create_relationships.rb]: 2 dòng

File: app/models/relationship.rb (hunk 1 above)
Comment (ThangPhanCong) [inline app/models/relationship.rb]: :p

**Pull request merged**